docs: various updates (#8652)
Homebrew/brew: docs/Analytics.md

## What?
Homebrew's analytics record some shared information for every event:

> - The Homebrew user agent, e.g. `Homebrew/0.9.9 (Macintosh; Intel macOS 10.12.0) curl/7.43.0`.

Reviewer: Nit: As we're changing this line to add a full stop, can we update the user agent string to something that's not ancient? 💀

Author: Interestingly, the user agent is listed as OS X, not macOS:

```console
% brew ruby -e 'puts HOMEBREW_USER_AGENT_CURL'
Homebrew/2.5.0-8-gac8282b (Macintosh; Intel Mac OS X 10.15.6) curl/7.64.1
```

Reviewer: Apparently this is intentional: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/brew.sh#L154-L155, and on my Catalina machine, yep, Safari still reports "Mac OS X".

- The [Google Analytics version](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#v), i.e. `1`.
- The Homebrew [analytics tracking ID](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#tid), e.g. `UA-75654628-1`.
- A Homebrew [analytics user ID](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#cid), e.g. `1BAB65CC-FE7F-4D8C-AB45-B7DB5A6BA9CB`. This is generated by `uuidgen` and stored in the repository-specific Git configuration variable `homebrew.analyticsuuid` within `$(brew --repository)/.git/config`. This does not allow us to track individual users, but does enable us to accurately measure user counts versus event counts. The ID is specific to the Homebrew package manager, and does not permit Homebrew maintainers to e.g. track you across websites you visit.
- If the [Google Analytics anonymous IP setting](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#aip) is enabled, i.e. `1`.
- The Homebrew [application name](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#an), e.g. `Homebrew`.

> - The Homebrew [application version](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#av), e.g. `0.9.9`.

Reviewer: ```suggestion
- The Homebrew [application version](https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters#av), e.g. `2.5.0`.
```

Homebrew's analytics are sent throughout Homebrew's execution to Google Analytics over HTTPS.

## Who?

> Summaries of installation and error analytics are publicly available [here](https://brew.sh/analytics/). A JSON API is also available. Homebrew maintainers are almost never granted more detailed analytics data beyond these public resources.

Reviewer 2: ```suggestion
Summaries of installation and error analytics are publicly available [here](https://brew.sh/analytics/). A JSON API is also available. The majority of Homebrew maintainers are not granted more detailed analytics data beyond these public resources.
```
